main.tf

```
@@ -1,5 +1,6 @@
 locals {
   instance_count = module.this.enabled ? 1 : 0
+  volumes_count  = var.ebs_volume_count != 0 && local.instance_count != 0 ? var.ebs_volume_count : 0
```

```suggestion
  volumes_count  = var.ebs_volume_count > 0 && local.instance_count > 0 ? var.ebs_volume_count : 0
```

Checking for `var.ebs_volume_count` and `local.instance_count` to be greater than 0 is more robust than directly checking if they are not equal to 0, since it's possible for `var.ebs_volume_count` to be a negative number.

---

Can imagine I did that initially but then I've seen the `!=` was used into another statement and I've changed it :D